### PR TITLE
Try no-print grammar first in Python 2

### LINKED ISSUE #752: black inserts unnecessary whitespace around print() with target_version = py27
black is adding whitespace between `print` and `(` with `--target-version py27`.

**Test file:**

```python
from __future__ import print_function

print("hello world")
```

**Result:**

```console
$ black t.py --target-version py27 --diff
--- t.py	2019-03-14 22:09:34.902887 +0000
+++ t.py	2019-03-14 22:10:21.022372 +0000
@@ -1,4 +1,4 @@
 from __future__ import print_function

-print("hello world")
+print ("hello world")

reformatted t.py
All done! ✨ 🍰 ✨
1 file reformatted.
```

Operating system: OS X 
Python version: 3.7.2
*Black* version: 19.3b0
Does also happen on master: yes

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Fixes #752

### CODE REVIEW

File: tests/data/python2_unicode_literals.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 u'hello'
 U"hello"
 Ur"hello"
+print("no spaces here please")
```
Comment (rouge8) [inline tests/data/python2_unicode_literals.py]: I think this test actually passes on master! It's only when specifying `py27` as a target version that it fails

Comment (rouge8) [reply tests/data/python2_unicode_literals.py]: Here's a failing test:

```diff
diff --git a/tests/data/python2_print_function.py b/tests/data/python2_print_function.py
new file mode 100755
index 0000000..c4e58c1
--- /dev/null
+++ b/tests/data/python2_print_function.py
@@ -0,0 +1,14 @@
+#!/usr/bin/env python2
+from __future__ import print_function
+
+print('hello')
+print(u'hello')
+
+# output
+
+
+#!/usr/bin/env python2
+from __future__ import print_function
+
+print("hello")
+print(u"hello")
diff --git a/tests/test_black.py b/tests/test_black.py
index 7c99f00..645eec7 100644
--- a/tests/test_black.py
+++ b/tests/test_black.py
@@ -461,6 +461,14 @@ class BlackTestCase(unittest.TestCase):
         # black.assert_equivalent(source, actual)
         black.assert_stable(source, actual, black.FileMode())
 
+    @patch("black.dump_to_file", dump_to_stderr)
+    def test_python2_print_function(self) -> None:
+        source, expected = read_data("python2_print_function")
+        mode = black.FileMode(target_versions={black.TargetVersion.PY27})
+        actual = fs(source, mode=mode)
+        self.assertFormatEqual(expected, actual)
+        black.assert_stable(source, actual, mode)
+
     @patch("black.dump_to_file", dump_to_stderr)
     def test_python2_unicode_literals(self) -> None:
         source, expected = read_data("python2_unicode_literals")
```

Comment (rouge8) [reply tests/data/python2_unicode_literals.py]: opened as https://github.com/ambv/black/pull/754

Comment (ambv) [reply tests/data/python2_unicode_literals.py]: Yeah, the space is actually only a dead giveaway of the fact that Black is treating the `print` as a statement. So, add multiple tests that exercise some variants of what's possible with print:
```
print
print(a, file=sys.stderr)
print >>sys.stderr, "text"
print hey,
def print():
  ...
```

Comment (rouge8) [reply tests/data/python2_unicode_literals.py]: I think most of those are already captured by tests/data/python2.py. I updated #754 with `print(a, file=sys.stderr)` though

**Comment by JelleZijlstra** [comment]
#754 is better

**Pull request closed without merging**